src/backend/metal/src/device.rs

```diff
@@ -133,19 +112,47 @@ pub struct Device {
     pub(crate) device: metal::Device,
     private_caps: PrivateCapabilities,
     queue: Arc<command::QueueInner>,
+    memory_types: [hal::MemoryType; 4],
 }
 unsafe impl Send for Device {}
 unsafe impl Sync for Device {}
 
-pub struct PhysicalDevice(pub(crate) metal::Device);
+pub struct PhysicalDevice {
+    raw: metal::Device,
+    memory_types: [hal::MemoryType; 4],
+}
 
 impl PhysicalDevice {
+    pub(crate) fn new(raw: metal::Device) -> Self {
+        PhysicalDevice {
+            raw,
+            memory_types: [
+                hal::MemoryType {
+                    properties: Properties::CPU_VISIBLE | Properties::CPU_CACHED,
+                    heap_index: 0,
+                },
+                hal::MemoryType {
```

Comment:
why is a second memory type here, exposing the same things as the first one?

Reply:
this was just copied from the previous code. Perhaps @grovesNL remembers?

Reply:
It was added in #1687 (https://github.com/gfx-rs/gfx/pull/1687/files#diff-97544b45a9d25f924c3a661a46b1cd0fR63). I guess it's not important unless there is some special meaning for these four indices?

Reply:
It doesn't seem to be important unless those indices have special meaning elsewhere? These properties became the same in #1571 with the removal of the `memory::WRITE_COMBINED` (https://github.com/gfx-rs/gfx/pull/1571/files#diff-97544b45a9d25f924c3a661a46b1cd0fL120)